src/v2i-hub/MobilityOperationPlugin/src/MobilityOperationPlugin.cpp

```diff
@@ -233,12 +231,166 @@ void MobilityOperationPlugin::createMobilityOperationXml( ptree &mobilityOperati
 	header.put("planId", "00000000-0000-0000-0000-000000000000");
-	header.put("timestamp","0000000000000000000");
+	header.put("timestamp", "0000000000000000000");
 	message.put_child("header", header);
 	message.put_child("body",body);
 	mobilityOperationXml.put_child( "TestMessage03", message);
+	return mobilityOperationXml;
 }
 
+/**
+ * Retreive next action from freight table using action_id
+ * 
+ * @param action_id string
+ */
+MobilityOperationPlugin::PortDrayage_Object MobilityOperationPlugin::retreiveNextAction( std::string action_id ) {
+	try{
+		// Set action_id
+		next_action_id->setString(1,action_id);
+		// Get current action
+		sql::ResultSet *cur_action = next_action_id->executeQuery();
+		if ( cur_action->first() ) {
+			FILE_LOG(logERROR) << "Column next_action: " << cur_action->getString("next_action") << std::endl;
+			// Set action_id to next_action
+			next_action->setString(1,cur_action->getString("next_action"));
+			// Retreive next action
+			sql::ResultSet *cur_action = next_action->executeQuery();
+			// Create PortDrayage_Object
+			PortDrayage_Object *rtn = new PortDrayage_Object();
+			if ( cur_action->first() )  {
+				rtn->cmv_id = cur_action->getInt("cmv_id");
+				rtn->cargo = cur_action->getBoolean("cargo");
+				rtn->operation = cur_action->getString("operation");
+				rtn->action_id = cur_action->getString("action_id");
+				// If cargo present, retreive cargo id
+				if ( rtn->cargo ) {
+					rtn->cargo_id = cur_action->getString("cargo_id");
+				}  
+				rtn->destination_long = cur_action->getInt("destination_long");
+				rtn->destination_lat =  cur_action->getInt("destination_lat");
+				
+				FILE_LOG(logERROR) << "Port Drayage Message : " << std::endl << 
+					"cmv_id : " << rtn->cmv_id << std::endl <<
+					"cargo_id : " << rtn->cargo_id << std::endl <<
+					"cargo : " << rtn->cargo << std::endl <<
+					"operation : " << rtn->operation << std::endl <<
+					"location_lat : " << rtn->location_lat << std::endl <<
+					"location_long : " << rtn->location_long << std::endl <<
+					"destination_lat : " << rtn->destination_lat << std::endl <<
+					"destination_long : " << rtn->destination_long << std::endl <<
+					"action_id : " << rtn->action_id << std::endl <<
+					"next_action : " << rtn->next_action << std::endl;
+				return *rtn;
+			}
+
+		}
+		else {
+			FILE_LOG(logERROR) << "No action with id : " << action_id << " found!";
+		}
+		
+	}
+	catch ( sql::SQLException &e ) {
+		FILE_LOG(logERROR) << "Error occurred during MYSQL Connection " << std::endl << e.what() << std::endl;
+		FILE_LOG(logERROR) << "Error code " << e.getErrorCode() << std::endl;
+		FILE_LOG(logERROR) << "Error status " << e.getSQLState() << std::endl;
+	}
+	
+}
+
+/**
+ * Retreive first action from first_action table using cmv_id.
+ * 
+ * @param cmv_id 
+ * @return PortDrayage_Object of first action 
+ */
+MobilityOperationPlugin::PortDrayage_Object MobilityOperationPlugin::retreiveFirstAction( uint32_t cmv_id ) {
```

Comment:
Typo for the word 'retrieve'. Please fix here and in the other locations within the code included in this PR as well.